blueprint/core/blueprint_EcmascriptEngine.cpp

```diff
@@ -342,12 +438,22 @@ namespace blueprint
                 for (int i = 0; i < nargs; ++i)
                     args.push_back(engine->readVarFromDukStack(engine->dukContext, i));
 
+                juce::var result;
+
                 // Now we can invoke the user method with its arguments
-                auto result = std::invoke(helper->callback, juce::var::NativeFunctionArgs(
-                    juce::var(),
-                    args.data(),
-                    static_cast<int>(args.size())
-                ));
+                try
+                {
+                    result = std::invoke(helper->callback, juce::var::NativeFunctionArgs(
+                        juce::var(),
+                        args.data(),
+                        static_cast<int>(args.size())
+                    ));
+                }
+                catch (Error& err)
+                {
+                    duk_push_error_object(ctx, DUK_ERR_TYPE_ERROR, err.what());
+                    return duk_throw(ctx);
```

Comment:
I'm not sure we want to return the `duk_ret_t` here? Maybe

```
(void) duk_throw(ctx);
return juce::var::undefined();
```

?

Reply:
Ah, let me check how the casting shakes out.  If it turns out the same I'm cool with making it explicit anyway, I've yet to see that being a bad idea.

Reply:
Sorry I misunderstood, but on further inspection this basically looks right as is.  I'm not a duktape expert but I think a negative `duk_ret_t` like `DUK_ERR_TYPE_ERROR` will be treated by the engine as throwing and never returning.  We are returning `duk_ret_t`, and using the `juce::var` only when not undefined, pushing to the stack and returning 1 to indicate there's a return value.

Reply:
Sorry I misunderstood, but on further inspection this basically looks right as is.  I'm not a duktape expert but I think a negative `duk_ret_t` like `DUK_ERR_TYPE_ERROR` will be treated by the engine as throwing and never returning.  We are returning `duk_ret_t`, and using the `juce::var` only when not undefined, pushing to the stack and returning 1 to indicate there's a return value.